woodwork/deserialize_accessor.py

```diff
@@ -0,0 +1,176 @@
+    if table_type == 'dask':
+        DASK_ERR_MSG = (
+            'Cannot load Dask DataFrame - unable to import Dask.\n\n'
+            'Please install with pip or conda:\n\n'
+            'python -m pip install "woodwork[dask]"\n\n'
+            'conda install dask'
+        )
+        lib = import_or_raise('dask.dataframe', DASK_ERR_MSG)
+    elif table_type == 'koalas':
+        KOALAS_ERR_MSG = (
+            'Cannot load Koalas DataFrame - unable to import Koalas.\n\n'
+            'Please install with pip or conda:\n\n'
+            'python -m pip install "woodwork[koalas]"\n\n'
+            'conda install koalas\n\n'
+            'conda install pyspark'
+        )
+        lib = import_or_raise('databricks.koalas', KOALAS_ERR_MSG)
+        compression = str(compression)
+    else:
+        lib = pd
+
+    if load_format == 'csv':
+        dataframe = lib.read_csv(
+            file,
+            engine=kwargs['engine'],
+            compression=compression,
+            encoding=kwargs['encoding'],
+        )
+    elif load_format == 'pickle':
+        dataframe = pd.read_pickle(file, **kwargs)
+    elif load_format == 'parquet':
+        dataframe = lib.read_parquet(file, engine=kwargs['engine'])
+
+    logical_types = {}
+    semantic_tags = {}
+    column_descriptions = {}
+    column_metadata = {}
+    for col in table_typing_info['column_typing_info']:
+        col_name = col['name']
+
+        ltype_metadata = col['logical_type']
+        ltype = ww.type_system.str_to_logical_type(ltype_metadata['type'], params=ltype_metadata['parameters'])
+
+        tags = col['semantic_tags']
+
+        if 'index' in tags:
+            tags.remove('index')
+        elif 'time_index' in tags:
+            tags.remove('time_index')
+
+        logical_types[col_name] = ltype
+        semantic_tags[col_name] = tags
+        column_descriptions[col_name] = col['description']
+        column_metadata[col_name] = col['metadata']
+
+    dataframe.ww.init(
+        name=table_typing_info.get('name'),
+        index=table_typing_info.get('index'),
+        time_index=table_typing_info.get('time_index'),
+        logical_types=logical_types,
+        semantic_tags=semantic_tags,
+        use_standard_tags=False,
+        table_metadata=table_typing_info.get('table_metadata'),
+        column_metadata=column_metadata,
+        column_descriptions=column_descriptions)
+
+    return dataframe
+
+
+def read_woodwork_table(path, profile_name=None, **kwargs):
+    '''Read Woodwork table from disk, S3 path, or URL.
+
+        Args:
+            path (str): Directory on disk, S3 path, or URL to read `table_description.json`.
+            profile_name (str, bool): The AWS profile specified to write to S3. Will default to None and search for AWS credentials.
+                Set to False to use an anonymous profile.
+            kwargs (keywords): Additional keyword arguments to pass as keyword arguments to the underlying deserialization method.
+    '''
```

Comment:
Maybe we should add a return type here as well, same as  in `typing_information_to_woodwork_table`, since this is part of the public API.

Reply:
Makes sense!

Reply:
makes sense to me! 

Now I'm thinking about which of these functions should be public, though. 

Having all of them present feels like it detracts from the useful ones (though `read_woodwork_table` is the only one that should get used with any regularity anyway, so maybe they all should be private but that one).

Reply:
Yeah, I think `read_woodwork_table` is the most important one. We could review all our functions and methods as part of #604 though.